### PR TITLE
[docs]Update Meshery Contributing Doc

### LINKED ISSUE #1564: [Docs] Update Meshery Contributing Doc
**Current State:**
The CONTRIBUTING.md is missing the caveat about Jekyll production & dev env amongst others.

**Desired State:**

[ ] Add [this caveat as seen in line 71](https://github.com/layer5io/service-mesh-performance/commit/a8af02878eb4711ccadb2e4fe01dad7b65a5b1b3#diff-6a3371457528722a734f3c51d9238c13)  to the [CONTRIBUTING.md](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow) Documentation Contribution Flow section.

[ ] Add caveat to [serve the site](https://github.com/layer5io/meshery/tree/master/docs#3-serve-the-site) section in README.

[ ] Update README [Set up your development environment](https://github.com/layer5io/meshery/tree/master/docs#1-set-up-your-development-environment) section as seen in line [49-54](https://github.com/layer5io/service-mesh-performance/commit/a8af02878eb4711ccadb2e4fe01dad7b65a5b1b3#diff-6a3371457528722a734f3c51d9238c13).

---
**Contributor Resources**
- [Meshery documentation site](https://meshery.layer5.io/docs/)
- [Meshery documentation source](https://github.com/layer5io/meshery/tree/master/docs)
- [Instructions for contributing to documentation](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)

### PR DESCRIPTION
**Pull request by Harshita-Kanal**
Signed-off-by: Harshita-Kanal <harshita.kgv@gmail.com>

**Description**

This PR fixes #1564 

**Notes for Reviewers**
Updated Meshery Contributing Doc

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -65,6 +65,7 @@ Please contribute! Meshery documentation uses GitHub Pages to host the docs site
 `vi <specific page>.md`
 1. Run site locally to preview changes.
 `make site`
+* **Note:** *From the Makefile, this command is actually running `$ bundle exec jekyll serve --drafts --livereload`. There are two Jekyll configuration, `jekyll serve` for developing locally and `jekyll build` when deploying for production based on differences of the handling of trailing slashes between Jekyll and GitHub Pages.*
```
Comment (kushthedude) [inline CONTRIBUTING.md:68]: This line is not appropriate, jekyll build is used by production parsers which simply generates the output of jekyll serve in simple html+js. Mentioning about jekyll build as production server is completely wrong.
I won't recommend such controversial statement to be added in the contributing file, else the below added description for WSL is great.

File: docs/README.md [hunk 2]
```diff
@@ -36,7 +40,7 @@ Note: Windows users can run Jekyll by following the [Windows Installation Guide]
   ```bash
   $ make site
   ```
-  *Note: From the Makefile, this command is actually running `$ bundle exec jekyll serve --drafts --livereload`*
+  *Note: From the Makefile, this command is actually running `$ bundle exec jekyll serve --drafts --livereload`. There are two Jekyll configuration, `jekyll serve` for developing locally and `jekyll build` when deploying for production based on differences of the handling of trailing slashes between Jekyll and GitHub Pages.*
```
Comment (kushthedude) [inline docs/README.md:43]: Please see above

**Review by kushthedude — COMMENTED** [review]
Hi @Harshita-Kanal , great work here, just a few suggestions!

File: docs/README.md (hunk 2 above)
Comment (Harshita-Kanal) [reply docs/README.md:43]: Thanks, I will remove the line added

Comment (kushthedude) [reply docs/README.md:43]: Let's do one thing, make it 
```suggestion
  *Note: From the Makefile, this command is actually running `$ bundle exec jekyll serve --drafts --livereload`. There are two Jekyll configuration, `jekyll serve` for developing locally and `jekyll build` when you need to generate the site artifacts for production.*
```

**Review by kushthedude — APPROVED** [review]
Great work @Harshita-Kanal 🎉 🎉 !!

**Comment by Harshita-Kanal** [comment]
Thanks! :smiley:

**Comment by kushthedude** [comment]
@Harshita-Kanal , thank you!

You're most welcome to join the Layer5 community. It's a welcoming bunch - http://slack.layer5.io. There's plenty more where that came from - [help-wanted](https://github.com/issues?utf8=✓&q=is%3Aopen+is%3Aissue+archived%3Afalse+org%3Alayer5io+label%3A%22help+wanted%22+).

**Comment by leecalcote** [comment]
Excellent!

**Pull request merged**